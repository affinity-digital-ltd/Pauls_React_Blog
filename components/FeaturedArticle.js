import { Component } from 'react'

import ArticlePreview from '../components/ArticlePreview'

class FeaturedArticle extends Component {
  render () {
    const { article } = this.props

    return <section className='c-featured'>
      <div className='o-featured__container'>
        <h2 className='c-featured__header'>Featured Article</h2>
        <div className='c-headers__divider--featured' />
        <div className='o-article-preview__container--featured'>
          <div className='c-article-preview__featured'>
            <ArticlePreview article={article} />
          </div>
        </div>
      </div>
    </section>
  }
}

export default FeaturedArticle
