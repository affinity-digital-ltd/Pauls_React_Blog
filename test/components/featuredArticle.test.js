import FeaturedArticle from '../../components/FeaturedArticle'

describe('<FeaturedArticle />', () => {
  it('it renders correctly', () => {
    const article = {
      'title': 'New Article #3',
      'image': '/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg'
    }

    const component = shallow(
      <FeaturedArticle article={article} />
    )

    expect(component).toMatchSnapshot()
  })
})