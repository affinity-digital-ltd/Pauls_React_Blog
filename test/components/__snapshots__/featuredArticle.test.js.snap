// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedArticle /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FeaturedArticle
    article={
      Object {
        "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
        "title": "New Article #3",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="o-featured__container"
      >
        <h2
          className="c-featured__header"
        >
          Featured Article
        </h2>
        <div
          className="c-headers__divider--featured"
        />
        <div
          className="o-article-preview__container--featured"
        >
          <div
            className="c-article-preview__featured"
          >
            <ArticlePreview
              article={
                Object {
                  "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                  "title": "New Article #3",
                }
              }
            />
          </div>
        </div>
      </div>,
      "className": "c-featured",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="c-featured__header"
          >
            Featured Article
          </h2>,
          <div
            className="c-headers__divider--featured"
          />,
          <div
            className="o-article-preview__container--featured"
          >
            <div
              className="c-article-preview__featured"
            >
              <ArticlePreview
                article={
                  Object {
                    "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                    "title": "New Article #3",
                  }
                }
              />
            </div>
          </div>,
        ],
        "className": "o-featured__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Featured Article",
            "className": "c-featured__header",
          },
          "ref": null,
          "rendered": "Featured Article",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "c-headers__divider--featured",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="c-article-preview__featured"
            >
              <ArticlePreview
                article={
                  Object {
                    "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                    "title": "New Article #3",
                  }
                }
              />
            </div>,
            "className": "o-article-preview__container--featured",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ArticlePreview
                article={
                  Object {
                    "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                    "title": "New Article #3",
                  }
                }
              />,
              "className": "c-article-preview__featured",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "article": Object {
                  "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                  "title": "New Article #3",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="o-featured__container"
        >
          <h2
            className="c-featured__header"
          >
            Featured Article
          </h2>
          <div
            className="c-headers__divider--featured"
          />
          <div
            className="o-article-preview__container--featured"
          >
            <div
              className="c-article-preview__featured"
            >
              <ArticlePreview
                article={
                  Object {
                    "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                    "title": "New Article #3",
                  }
                }
              />
            </div>
          </div>
        </div>,
        "className": "c-featured",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="c-featured__header"
            >
              Featured Article
            </h2>,
            <div
              className="c-headers__divider--featured"
            />,
            <div
              className="o-article-preview__container--featured"
            >
              <div
                className="c-article-preview__featured"
              >
                <ArticlePreview
                  article={
                    Object {
                      "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                      "title": "New Article #3",
                    }
                  }
                />
              </div>
            </div>,
          ],
          "className": "o-featured__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Featured Article",
              "className": "c-featured__header",
            },
            "ref": null,
            "rendered": "Featured Article",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-headers__divider--featured",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="c-article-preview__featured"
              >
                <ArticlePreview
                  article={
                    Object {
                      "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                      "title": "New Article #3",
                    }
                  }
                />
              </div>,
              "className": "o-article-preview__container--featured",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <ArticlePreview
                  article={
                    Object {
                      "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                      "title": "New Article #3",
                    }
                  }
                />,
                "className": "c-article-preview__featured",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "article": Object {
                    "image": "/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBEUT09IiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--7d12c66030ae5e39faf3acc1b6e9cd19139f8dc6/water-tap-black-and-white-macro-615326.jpeg",
                    "title": "New Article #3",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
